File: server/src/protocol/BlindsStatusBuilder.cpp
#include "BlindsStatusBuilder.h"
#include "BlindsStatusCommand.h"

#include <utility/BytesConverter.h>

using namespace std;
using namespace utility;
using namespace communication;

BlindsStatusBuilder::BlindsStatusBuilder()
{}

BlindsStatusBuilder::~BlindsStatusBuilder()
{}

unique_ptr<Command> BlindsStatusBuilder::create(const vector<uint8_t> &commandInBytes)
{
    auto command = make_unique<BlindsStatusCommand>();

    return move(command);
}

File: mobile_client/app/src/main/java/wpam/mobile_client/protocol/BlindsStatusBuilder.java
package wpam.mobile_client.protocol;

import java.io.Closeable;
import java.util.ArrayList;

public final class BlindsStatusBuilder implements CommandBuilder, Closeable
	{
		public BlindsStatusBuilder()
		{
		}
		public final void close()
		{
		}

		@Override
		public Command create(ArrayList<Integer> commandInBytes)
		{
			Command command  = new BlindsStatusCommand();
			return command;
		}
	}